src/components/pages/AdminAddCourse/AdminAddCoursesForm.js

```diff
@@ -1,181 +1,400 @@
+const rangeConfig = {
+  rules: [
+    {
+      type: 'array',
+      message: 'Please select time!',
+    },
+  ],
 };
 
 function AdminAddCoursesForm(props) {
   const [formValues, setFormValues] = useState(initialFormValues);
-  const [formPreReqs, setFormPreReqs] = useState({ prereq: '' });
-  const { authState, oktaAuth } = useOktaAuth();
+  const { courseinfo } = props;
+  const [isModalVisible, setIsModalVisible] = useState(false);
   const dispatch = useDispatch();
+  const idToken = useOktaAuth().oktaAuth.getIdToken();
 
-  useEffect(() => {
-    if (authState !== null) {
-      if (authState.isAuthenticated !== false) {
-        dispatch(getCurrentUser(authState.idToken.idToken, oktaAuth));
-      }
-    }
-  }, []);
-
-  let history = useHistory();
-
-  //currently being blocked from the BE due to only a instructor can add courses.. BE middleware will need to be added for admin.
-  function handleSubmit(e) {
-    e.preventDefault();
-    axiosWithAuth(authState.idToken.idToken)
-      .post('/courses', formValues)
-      .then(() => {
-        history.push('/admin-courses');
-      })
-      .catch(err => {
-        console.error(err);
-      });
-    const merged = {
-      ...formValues,
-      prereq: placeHolder,
-    };
-    program_list.push(props.addClass(merged).payload);
-    setFormValues(initialFormValues);
-    clearPrereq();
-  }
+  const daysOfWeek = [];
 
-  const handleChange = e => {
-    if (e.target.name !== 'prereq') {
+  const showModal = () => {
+    setIsModalVisible(true);
+    if (props.button_name === 'Edit Course') {
       setFormValues({
-        ...formValues,
-        [e.target.name]: e.target.value,
-      });
-    } else if (e.target.name === 'prereq') {
-      setFormPreReqs({
-        ...formPreReqs,
-        [e.target.name]: e.target.value,
+        course_id: courseinfo.course_id,
+        course_name: courseinfo.course_name,
+        course_description: courseinfo.course_description,
+        days_of_week: courseinfo.days_of_week,
+        max_size: courseinfo.max_size,
+        max_age: courseinfo.max_age,
+        min_age: courseinfo.min_age,
+        start_date: courseinfo.start_date,
+        end_date: courseinfo.end_date,
+        start_time: courseinfo.start_time,
+        end_time: courseinfo.end_time,
+        number_of_sessions: courseinfo.number_of_sessions,
+        instructor_id: courseinfo.instructor_id,
       });
     }
   };
 
-  const addPrereq = () => {
-    if (formPreReqs.prereq.length !== 0) {
-      formPrerequisite = formPreReqs.prereq;
-      placeHolder.push(formPrerequisite);
-      buildString(formPrerequisite);
+  const handleClose = () => {
+    setIsModalVisible(false);
+  };
+
+  const handleSubmit = e => {
+    if (props.button_name === 'Add Course') {
+      dispatch(addCourse(idToken, formValues));
+    } else {
+      dispatch(editCourse(idToken, formValues));
+      alert('Edits Submitted!');
     }
+
+    setIsModalVisible(false);
+    window.location.reload();
```

Comment:
Curious as to what this is doing for the submit function? Reloading the entire web page? Is it to update the state?

Reply:
its so that the new course will be displayed on the page